linera-indexer/src/graphql.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use graphql_client::GraphQLQuery;
+use linera_base::{
+    crypto::CryptoHash,
+    data_types::{BlockHeight, Timestamp},
+    identifiers::{ChainId, Destination, Owner},
+};
+use linera_chain::data_types::{
+    notifications::Notification, Event, ExecutedBlock, HashedValue, IncomingMessage, Origin,
+    OutgoingMessage,
+};
+use linera_execution::{committee::Epoch, Message, Operation};
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/notifications.graphql",
+    response_derives = "Debug"
+)]
+pub struct Notifications;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/block.graphql",
+    response_derives = "Debug"
+)]
+pub struct Block;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../linera-explorer/graphql/schema.graphql",
+    query_path = "../linera-explorer/graphql/chains.graphql",
+    response_derives = "Debug"
+)]
+pub struct Chains;
+
+impl From<block::BlockBlockValueExecutedBlockBlockIncomingMessages> for IncomingMessage {
+    fn from(val: block::BlockBlockValueExecutedBlockBlockIncomingMessages) -> Self {
+        IncomingMessage {
+            origin: val.origin,
+            event: val.event,
+        }
+    }
+}
+
+impl From<block::BlockBlockValueExecutedBlockBlock> for linera_chain::data_types::Block {
+    fn from(val: block::BlockBlockValueExecutedBlockBlock) -> Self {
+        let incoming_messages: Vec<IncomingMessage> = val
+            .incoming_messages
+            .into_iter()
+            .map(|b: block::BlockBlockValueExecutedBlockBlockIncomingMessages| b.into())
```

Comment:
Is that type annotation needed? (Also below.)

```suggestion
            .map(IncomingMessage::from)
```